main_test.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestExtractNpmVersion(t *testing.T) {
+
+	testCases := []struct {
+		pkgJSON string
+		want    string
+		wantE   error
+	}{
+		{`{"engines":{"npm":"3.0.1"}}`, "3.0.1", nil},
+		{`"engines":{"npm":"3.0.1"}}`, "", errors.New("")},
```

Comment:
do not use errors.New("") to represent that this case should fail, either test for the desired error message or use a boolean to differentiate the case when error is expected